Added support for building via makedeps/make target

zoltrain opened the pull request:
- Rather then putting dependencies into the metallus.yml file,
  dependency installion for build can be handed off to make

Commit:
Added support for building via makedeps/make target

 - Rather then putting dependencies into the metallus.yml file,
   dependency installion for build can be handed off to make

dwb commented on metallus/command.py:
```diff
@@ -113,10 +113,14 @@ def _build(self, args):
+        else:
+            dockerfile = AptDockerFile(self.project,
+                           self.project.current_job.apt_repos,
+                           self.project.current_job.apt_keys)
```
I would have this out in a module function in `dockerfile`, and `from . import dockerfile` above.

dwb commented on metallus/dockerfile.py:
```diff
@@ -172,31 +197,33 @@ def _install_packages(self, config):
-    def _write_line(self, l):
-        self.contents.write(l.strip() + "\n")
+        self._flush_commands()
+        src_path = self.path_dir + "/src"
```
use `os.path.join`

dwb commented on metallus/dockerfile.py:
```diff
@@ -172,31 +197,33 @@ def _install_packages(self, config):
+        if os.path.exists(src_path):
+            shutil.rmtree(src_path)
+
+        shutil.copytree(self.project.source.path + "/", self.path_dir + "/src")
```
and here

Commit:
Added recusive expansion of makefile includes

 - this is the foundation for doing smart rebuilds based on the makefile
   context, we could bake into the tags of the build-deps image a hash
   of all the makefile context objects, giving a unique fileset. if that
   changes then we can rebuild the image
